src/shared/components/VideoTileBase/VideoTileCover.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useState } from 'react'
+import { CSSTransition, SwitchTransition } from 'react-transition-group'
+import useResizeObserver from 'use-resize-observer'
+
+import {
+  SvgGlyphClose,
+  SvgGlyphDraft,
+  SvgGlyphHide,
+  SvgIllustrativeReupload,
+  SvgLargeEdit,
+  SvgLargeUploadFailed,
+  SvgOutlineVideo,
+} from '@/shared/icons'
+import { transitions } from '@/shared/theme'
+import { formatDurationShort } from '@/utils/time'
+
+import { PullUp } from './PullUp'
+import { Anchor } from './VideoTileBase.styles'
+import {
+  CoverContainer,
+  CoverDurationOverlay,
+  CoverHoverOverlay,
+  CoverIconWrapper,
+  CoverImage,
+  CoverImageContainer,
+  CoverNoImage,
+  CoverSkeletonLoader,
+  CoverThumbnailUploadFailed,
+  CoverTopLeftContainer,
+  CoverVideoPublishingStateOverlay,
+  CoverWrapper,
+  DELAYED_FADE_CLASSNAME,
+  PublishingStateText,
+  RemoveButton,
+  SkeletonHoverOverlay,
+  UploadProgressTransition,
+} from './VideoTileCover.style'
+
+import { VideoTileBaseMetaProps, VideoTilePublisherProps } from '.'
```

Comment:
This will be a circular dependency right? Does `VideoTileCoverProps` need all the other props?